### PR TITLE
Create a modal for user profile information #22

### LINKED ISSUE #22: Create a modal for user profile information
* There is a link for User Profile in the vertical dropdown 

- [ ] Create a modal that would pop up on clicking that link

- [ ] The modal will contain user profile information 

- [ ] The background (everything behind the modal) should either be dimmed or blurred - you can use z-index or animation

- [ ] Clicking anywhere outside the modal should close it

- [ ] The modal should also have a *X* symbol or a 'Close' button for closing it

### PR DESCRIPTION
**Pull request by anand-nakat**
Fixes #22 
1. Modal created using JQuery/Bootstrap 4. It displays currently logged in user information. 

image

2. Currently there is dummy data in the profile as Database doesn't exist. But I've created Pipeline to fetch & Update User Information from Database through ajax. By using ajax, we can communicate with database to fetch/update data, without refreshing the page ie Asynchronous calls to database.
3. It has option to edit profile data. The input fields are locked by default and can only be unlocked by clicking in corresponding edit button. Also The 'Save Changes' button is not clickable until the data is changed. 

image

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -409,9 +482,96 @@ function initMap() {
...
+                    arr=JSON.parse(response);
+                    
+                    //Insert the fetched data into modal input fields
+                    $("#full-name").val(arr[0]["name"]);
+                    $("#email").val(arr[0]["email"]);
+                    $("#organisation").val(arr[0]["organisation"]);
+                    $("#state").val(arr[0]["state"]);
+                  }
+                });
+ */
```
Comment (anand-nakat) [inline index.php:524]: Code to fetch data, if modal opened. It's been commented now due to database non availability.

File: index.php [hunk 2]
```diff
@@ -409,9 +482,96 @@ function initMap() {
...
+                  alert(response);
+                  $("input").prop('disabled', true);
+                  $("input").css('background-color','antiquewhite');
+                  $("#apply-changes-btn").prop("disabled",false);
+                  $("#apply-changes-btn").css({'background-color':'#6c757d','font-weight':'500','cursor': 'not-allowed'});
+                  $(".nav-username h2,.menu-username h2").html('Welcome '+name+'!');
+                  $(".modal-title").html(name+"'s Profile");
+                }
+              }); */
+
```
Comment (anand-nakat) [inline index.php:566]: Code to update the user profile, if clicked on 'Save Changes' Button. This has been commented as well

File: index.php [hunk 3]
```diff
@@ -409,9 +482,96 @@ function initMap() {
...
+          console.log("edit clciked");
+          $(this).prev().prop("disabled",false);
+          $(this).prev().css("background-color","aliceblue");
+          $(this).prev().focus();
+        });
+
+        $(".modal .modal-body input").change( function(){
+            $("#apply-changes-btn").prop("disabled",false);
+            $("#apply-changes-btn").css({'background-color':'#28a745','font-weight':'600','cursor': 'pointer'});
+        });
```
Comment (anand-nakat) [inline index.php:539]: If input is changed, then unlock SAVE CHANGES button

File: index.php [hunk 4]
```diff
@@ -409,9 +482,96 @@ function initMap() {
...
+$(document).ready( function () {  
+
+        const userId=<?php echo($_SESSION['user_id']); ?>;
+        
+        //Lock Input Field By Default as Modal is Opened    
+        $("#my-profile").click(function () {
+          $("input").prop('disabled', true);
+          $("input").css('background-color','antiquewhite');
+          // $(this).prev().css("background-color","antiquewhite");
+          $("#apply-changes-btn").css({'background-color':'#6c757d','font-weight':'500','cursor': 'not-allowed'});
```
Comment (anand-nakat) [inline index.php:504]: Lock input and Save Changes button, by default as modal is opened every time.

**Review by Chintan2108 — APPROVED** [review]
Great work @anand-nakat

**Comment by Chintan2108** [comment]
give @anand-nakat 75 points

**Pull request merged**